scene/mission-replay/src/main/java/com/esri/samples/scene/MissionController.java

```diff
@@ -0,0 +1,338 @@
+
+    // clear previous mission data
+    missionData = new ArrayList<>();
+
+    // get mission data
+    String mission = missionSelector.getSelectionModel().getSelectedItem();
+    missionData = getMissionData(mission);
+    animationModel.setFrames(missionData.size());
+    animationModel.setKeyframe(0);
+
+    // draw mission route on mini map
+    PointCollection points = new PointCollection(SpatialReferences.getWgs84());
+    points.addAll(missionData.stream().map(m -> (Point) m.get("position")).collect(Collectors.toList()));
+    Polyline route = new Polyline(points);
+    routeGraphic.setGeometry(route);
+
+    // refresh mini map zoom and show initial keyframe
+    mapView.setViewpointScaleAsync(100000).addDoneListener(() -> Platform.runLater(() -> animate(0)));
+    animation.stop();
+
+    // enable play button
+    playButton.setSelected(false);
+    playButton.setDisable(false);
+  }
+
+  /**
+   * Loads the mission data from a .csv file into memory.
+   *
+   * @param mission .csv file name containing the mission data
+   * @return ordered list of mapped key value pairs representing coordinates and rotation parameters for each step of
+   * the mission
+   */
+  private List<Map<String, Object>> getMissionData(String mission) {
+
+    // open a file reader to the mission file that automatically closes after read
+    try (BufferedReader missionFile = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream("/" + mission)))) {
+      List<Map<String, Object>> missionData = new ArrayList<>();
+      missionFile.lines()
+          .map(l -> l.split(","))
+          .map(l -> {
+            // create a map of parameters (ordinates) to values
+            Map<String, Object> ordinates = new HashMap<>();
+            ordinates.put("position", new Point(Double.valueOf(l[0]), Double.valueOf(l[1]), Double
+                .valueOf(l[2]), SpatialReferences.getWgs84()));
+            ordinates.put("heading", Double.valueOf(l[3]));
+            ordinates.put("pitch", Double.valueOf(l[4]));
+            ordinates.put("roll", Double.valueOf(l[5]));
+            return ordinates;
+          })
+          .collect(Collectors.toCollection(() -> missionData));
+
+      return missionData;
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    throw new RuntimeException("Error reading mission file: " + mission);
+  }
+
+  /**
+   * Animates a single keyframe corresponding to the index in the mission data profile. Updates the position and
+   * rotation of the 2D/3D plane graphic and sets the camera viewpoint.
+   *
+   * @param keyframe index in mission data to show
+   */
+  private void animate(int keyframe) {
+    // get the next position
+    Map<String, Object> datum = missionData.get(keyframe);
+    Point position = (Point) datum.get("position");
+
+    // update the model bean with new parameters
+    planeModel.setAltitude(position.getZ());
+    planeModel.setHeading((double) datum.get("heading"));
```

Comment:
The string constants can be moved out as class fields. I can be reused in lines 92-94.